graviti/utility/lazy2.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+#
+# Copyright 2022 Graviti. Licensed under MIT License.
+#
+
+"""Lazy list related class."""
+
+from itertools import chain
+from math import ceil
+from typing import Any, Callable, Iterator, List, Optional, Tuple, Union
+
+import pyarrow as pa
+
+
+class LazyList:
+    """LazyList is a lazy load list.
+
+    Arguments:
+        factory: The parent :class:`LazyFactory` instance.
+        keys: The keys to access the array from factory.
+
+    """
+
+    def __init__(
+        self,
+        factory: "LazyFactory",
+        keys: Tuple[str, ...],
+    ) -> None:
+        self._total_count = factory._total_count
+        self._keys = keys
+        self._limit = factory._limit
+        self._pages: List[Union["LazyPage", pa.Array]] = [
+            LazyPage(pos, self) for pos in range(factory._page_number)
+        ]
+        self._factory = factory
+
+    def __len__(self) -> int:
+        return self._total_count
+
+    def __iter__(self) -> Iterator[Any]:
+        return chain(*self._pages)
+
+    def __getitem__(self, index: int) -> Any:
```

Comment:
Is the return type `Any` or `pyarrow.Scalar`?

Reply:
It should be a TypeVar,
Mypy does not recognize pa.Array, we need a more compact typing system for LazyList, I will do it later.